src/libs/Network.js

```diff
@@ -50,9 +50,18 @@ function processNetworkRequestQueue() {
             return;
         }
 
-        const finalParameters = _.isFunction(enhanceParameters)
-            ? enhanceParameters(queuedRequest.command, queuedRequest.data)
-            : queuedRequest.data;
+        let finalParameters;
+
+        try {
+            finalParameters = _.isFunction(enhanceParameters)
+                ? enhanceParameters(queuedRequest.command, queuedRequest.data)
+                : queuedRequest.data;
+        } catch (err) {
+            // Something went wrong when enhancing parameters assume we should not
+            // make this request at all and clear the request queue
+            networkRequestQueue = [];
+            return;
+        }
```

Comment:
If someone adds another `throw` inside `enhanceParameters`, or there is some kind of bug that gets introduced that throws an unexpected error, it's not clear that this will cause the network queue to be cleared. I'd rather be very explicit about how this work by doing maybe one of these solutions:

1. Have a method `Network.clearRequestQueue()` that is called in place of the `throw new Error('A request was made without an authToken');`. I'd prefer this one because it's very clear what the intention of the code is.
2. Have specific checking here on `err` to only clear the request queue when the `A request was made without an authToken` error is thrown. I don't like this as much because if I were just looking at `addAuthTokenToParameters` I wouldn't know that clearing the network queue could be a side-effect of throwing an error.

Reply:
Ok, that's a valid concern. The problem with `1` if that we probably don't need to make any request if we have no `authToken`. Clearing the network request queue alone doesn't achieve this since we are _already_ making this request on the next line.

Reply:
Will think about this some more.

Reply:
Ok I got a better solution. We will:

1. Clear and pause the request queue via `addAuthTokenToParameters()`
2. Check to see if calling `enhanceParameters()` paused the queue and return early